src/api/app/controllers/trigger_controller.rb

```diff
@@ -91,9 +91,9 @@ def set_multibuild_flavor
     # @package is a String if @project has a project link, no need to do anything then.
     return unless @package.is_a?(Package)
 
-    # We use the package parameter if it is a valid multibuild flavor of the package
-    # See https://github.com/openSUSE/open-build-service/wiki/Links#mulitbuild-packages
-    @package = params[:package] if @package.multibuild_flavor?(params[:package])
+    # We use the package parameter if it is a multibuild flavor of a package
+    # The package object may not exist (different project, maybe even remote)
+    @package = params[:package] if params[:package].include?(':')
```

Comment:
At this point it must be, see line 92...

Reply:
it can't be if it is remote (and see line 60 which is broken anyway, since you do not know if the packages froms from remote, even when there is a project link to remote)

but the code seems also not to handle local instances correctly, but this way it works in general and does not need plenty if special cases.

Reply:
and you can not call multibuild_flavor? on a string neither.

Reply:
How can @package be a string after line 92?

Reply:
@package ||= params[:package]
  is setting it to a string, no?

Reply:
Only if `Package.get_by_project_and_name` above did not set it to a package object (there is a package in the DB in @project or in a project-link of @project) or bail out with an exception (in case there is no local package and we don't follow project links for this token).

Reply:
And not matter if some line above sets it to a string. If `return unless...` on line 92 will return unless it's a `Package` object...